added ability to whitelist properties for additing and supply additio…

jpeters5392 opened the pull request:
…nal text for the component title

Commit:
added ability to whitelist properties for additing and supply additional text for the component title

jpeters5392 commented on packages/react-live-props/src/ReactLiveProps/ReactLiveProps.js:
```diff
@@ -43,17 +55,26 @@ export default class ReactLiveProps extends Component {
 
+    const renderSchema = {
+      ...schema,
+      title: this.buildComponentTitle(schema.title, additionalTitleText)
```
The schema title is rendered by `react-jsonschema-form` at the top of the form.

jpeters5392 commented:
<img width="853" alt="screen shot 2019-02-02 at 9 45 04 am" src="https://user-images.githubusercontent.com/1538689/52166810-6f5d9300-26cf-11e9-95cb-1fbe2f247077.png">
<img width="837" alt="screen shot 2019-02-02 at 9 45 46 am" src="https://user-images.githubusercontent.com/1538689/52166812-71275680-26cf-11e9-9186-17a24eac3477.png">
<img width="839" alt="screen shot 2019-02-02 at 9 46 03 am" src="https://user-images.githubusercontent.com/1538689/52166813-72588380-26cf-11e9-8e7b-d43ca336eb69.png">

jpeters5392 commented on packages/react-live-props/src/ReactLiveProps/ReactLiveProps.js:
```diff
@@ -43,17 +55,26 @@ export default class ReactLiveProps extends Component {
           onChange={this._onChange}
         />
 
+        <hr />
```
I'm not sure if this is desired or not.  I just thought that it might need something to separate the component preview from the form.

kevindavi5 replied:
I like it.

natergj commented on packages/react-live-props/src/EditablePropsTable/EditablePropsTable.js:
```diff
@@ -11,15 +11,39 @@ export default class EditablePropsTable extends Component {
+    uiSchema: PropTypes.object
   }
 
+  static getDerivedStateFromProps(props, state) {
```
does schema need to be on state? The react docs seem to discourage use of this method unless necessary. https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

jpeters5392 replied:
Since this is technically creating a new object every time then it could be moved to the render method.  I was going to come back through and fix this up so it would not create a new object every time though and at that point it would not be suitable for the render method.

jpeters5392 commented:
I might have to revisit this.  Dropping the form in to the shadow DOM makes rendering arbitrary components in docz difficult since it means I either need to scrape all of the parent's CSS in to the shadow DOM, which brings back the original issue, or identify the global CSS from the component in the playground and copy just that in to the shadow DOM.

For reference, the original issue is that docz's root node has an HTML id of `root`, and the react-jsonschema-form library also mounts itself into a fieldset with an id of `root`, which blows up the CSS for our rendering.  I guess a short term hack would be to remove the shadow DOM and add some specific CSS that resets the `#root` that exists within our library.

jpeters5392 commented:
@transitive-bullshit The shadow-dom changes were a solution to CSS issues due to docz and react-jsonschema-form both creating DOM elements with the id of `root`.  I removed those for now and implemented a much smaller CSS hack to make it render correctly
